bin/codegen-timezones.php

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env hhvm
+<?hh
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace HH\Lib\Experimental\DateTime\_Private;
+
+use namespace HH\Lib\{Keyset, Regex, Str};
+use type Facebook\HackCodegen\{
+  CodegenFileType,
+  HackBuilderValues,
+  HackCodegenConfig,
+  HackCodegenFactory,
+};
+
+/**
+ * Generate the HH\Lib\Experimental\DateTime\Zone enum based on the timezones
+ * supported by the current HHVM version.
+ */
+<<__EntryPoint>>
+function codegen_timezones(): void {
+  $docblock = <<<END
+All supported timezones. This includes:
+
+- UTC
+- all supported tzdata timezones like "America/Los_Angeles"
+- all UTC offsets that at least one supported timezone uses for either its
+  winter time or its summer time
+
+We intentionally don't include common aliases like PST (Pacific Standard
+Time) because these can be confusing and ambiguous, e.g. the meaning of PST
+is unclear when dealing with dates that fall in PDT (Pacific Daylight Time)
+range. Instead, use the standard tzdata timezone like "America/Los_Angeles"
+which correctly resolves to either PST or PDT for any specified date; or
+you can use an explicit offset like UTC-08:00 which is always unambiguous.
```

Comment:
This is reasonable (and best practice), but we probably need these somewhere; separate enum?